app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedFragment.java

```diff
@@ -0,0 +1,216 @@
+package ru.vssemikoz.newsfeed.newsfeed;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+import java.util.Objects;
+
+import ru.vssemikoz.newsfeed.R;
+import ru.vssemikoz.newsfeed.adapters.NewsFeedAdapter;
+import ru.vssemikoz.newsfeed.dialogs.PickCategoryDialog;
+import ru.vssemikoz.newsfeed.models.Category;
+import ru.vssemikoz.newsfeed.models.NewsItem;
+import ru.vssemikoz.newsfeed.navigator.Navigator;
+import ru.vssemikoz.newsfeed.storage.IconicStorage;
+
+import static androidx.core.util.Preconditions.checkNotNull;
+
+public class NewsFeedFragment extends Fragment implements NewsFeedContract.View,
+        PickCategoryDialog.OnCategorySelectedListener {
+    private String TAG = NewsFeedFragment.class.getName();
+    private String CURRENT_CATEGORY = "CURRENT_CATEGORY";
+    private String CURRENT_SHOW_FAVORITE = "CURRENT_SHOW_FAVORITE";
+
+    private NewsFeedContract.Presenter presenter;
+    private Context context;
+    private NewsFeedAdapter adapter;
+    private RecyclerView recyclerView;
+    private TextView emptyView;
+    private ImageButton favoriteNewsButton;
+    private ImageButton categoryButton;
+    private ProgressBar progressBar;
+    private TextView descriptionView;
+
+    public NewsFeedFragment() {
+    }
+
+    static NewsFeedFragment newInstance() {
```

Comment:
There's no need in creating `newInstance()` method since each `Fragment` already has it.